[nicolengk] iP

nicolengk opened the pull request:
### Sara Pro X Flip 5000🐹🐁

> "People who love to eat are always the best people." - Julia Child ([source](https://pos.toasttab.com/blog/on-the-line/food-quotes-and-captions))

Having Sara is great, Sara frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,
1. _eat_
2. download it from [here](https://github.com/nicolengk/ip).
3. double-click it.
4. add your tasks.
5. let it manage your tasks for you 😉
And it is **FREE!**

Features: 
- [ ]  Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
 ---

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

nicolengk committed:
feat: added goodbye

nicolengk committed:
feat: Rename, Greet, Exit

nicolengk committed:
feat: echo

nicolengk committed:
feat: add, list

nicolengk committed:
level 2 fix

nicolengk committed:
feat: mark as done

nicolengk committed:
feat: todos,events,deadlines

nicolengk committed:
testing

nicolengk committed:
feat: handle errors

nicolengk committed:
handle error fix

nicolengk committed:
feat: delete

nicolengk committed:
seperating classes

nicolengk committed:
refactoring

nicolengk committed:
add save feature that save the tasks automatically

nicolengk committed:
Merge branch 'branch-Level-7'

nicolengk committed:
add feature to read date and time

nicolengk committed:
Merge branch 'branch-Level-8'

nicolengk committed:
refactor: create Ui Parser and TaskList classes

nicolengk committed:
add duke packages

nicolengk committed:
Merge commit '28ad2b878f1a21d03879e81cd441d0175d89e6b1'

nicolengk committed:
Add gradle support

nicolengk committed:
Add testing for Task and TaskList class

nicolengk committed:
add jar file

nicolengk committed:
Add javadocs to classes

nicolengk committed:
Fix code to comply with coding standards

nicolengk committed:
add feat: find task

nicolengk committed:
Merge branch 'branch-A-JavaDoc'

nicolengk committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/utils/Parser.java

nicolengk committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/task/Task.java

nicolengk committed:
Add javadocs for find feature

nicolengk committed:
Revert "Add javadocs for find feature"

This reverts commit c0ac313440df35efd02bf699e685287ebb2d085c.

nicolengk committed:
Add javadocs for find feature

vansh284 commented on src/test/java/duke/task/TaskTest.java:
```diff
@@ -0,0 +1,46 @@
+public class TaskTest {
+    @Test
+    public void markTaskDone_success() throws DukeException {
+        TaskList task = new TaskList();
```
Would the plural form be more appropriate for the naming of this variable since it is a collection of objects? I noticed the same issue in the other test methods as well.

vansh284 commented on src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,60 @@
+ * The `Storage` class handles reading and writing tasks to a data file.
+ */
+public class Storage {
+    private static final String DATA_FILE_PATH = "data.txt";
```
I like how you use all uppercase and underscores for constant names, really improves readability 😄 .

vansh284 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,116 @@
+    }
+
+    /**
+     * Creates a task object from a data string
```
An empty line between the description and parameter section of JavaDoc comments seems to be the practice recommended by our coding standard? I noticed this issue in other places in the code base as well.

vansh284 commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,132 @@
+     * Extracts the command keyword from the user input.
+     * @return The extracted command keyword in uppercase.
+     */
+    public String inputCommand() {
```
Would the code here be clearer if a verb was used to name the method instead? One possibility could be `getInputCommand`.

vansh284 reviewed (commented):
Overall, the code quality seems good, just a few nits to fix here and there. I especially like the way you organized your classes into packages, makes the codebase flow well.

nicolengk committed:
Change greeting logo

tim-pipi commented on data.txt:
Should this be included in `.gitignore` instead?

tim-pipi commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,66 @@
+     * @param taskData The data string containing deadline task information.
+     * @return A `Deadline` task object created from the data string, or `null` if the data is incomplete or invalid.
+     */
+    public static Deadline createDeadlineFromData(String taskData) {
+        String[] taskParts = taskData.split("\\s*\\|\\s*");
+
+        if (taskParts.length >= 3 && taskParts[0].trim().equals("D")) {
+            String doneStatus = taskParts[1].trim();
+            String description = taskParts[2].trim();
+
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+            LocalDateTime by = LocalDateTime.parse(taskParts[3].trim(), formatter);
+
+            Deadline deadline = new Deadline(description, by);
+            if (doneStatus.equals("1")) {
+                deadline.markDone();
+            }
+            return deadline;
+        }
+        return null; // incomplete data.txt
+    }
```
I think this function is a good place to do this! 👍

tim-pipi commented on src/main/java/duke/task/Task.java:
Actually, I was thinking if Task should be an abstract class instead since the user technically shouldn't be able to create a generic `Task` object.

tim-pipi commented on src/main/java/duke/utils/TaskList.java:
```diff
@@ -0,0 +1,161 @@
+ * The `TaskList` class represents a list of tasks and provides methods for managing tasks within the list.
+ */
+public class TaskList {
+    private List<Task> toDoList;
```
Perhaps naming it `taskList` is still better since `toDoList` might make people think that its a list of `ToDo`s instead of `Task`s?

tim-pipi reviewed (commented):
LGTM. Overall, your code is really using the OOP paradigm really well!

nicolengk committed:
Add gui

nicolengk committed:
Add gui

nicolengk committed:
Feat: add gui

nicolengk committed:
Merge branch 'branch-Level-10'

nicolengk committed:
Fix gradle

nicolengk committed:
Add checkstyle

nicolengk committed:
Fix styling issues

nicolengk committed:
Merge branch 'branch-A-CheckStyle'

nicolengk committed:
Add assertions

Currently, the code does not make use of any assertion. Adding assertions helps to improve code quality and ensure code reliability. Using assertions systematically will help catch issues early in the development process, making debugging and maintenance more manageable.

nicolengk committed:
Attempt to extract Message class

nicolengk committed:
Abstract Message class from Duke class
This helps to make the run method from the Duke class shorter and more readerable as it currently contains more than 30 lines of code.

nicolengk committed:
Merge pull request #1 from nicolengk/branch-A-Assertions

Add assertions

nicolengk committed:
Merge pull request #3 from nicolengk/branch-A-CodeQuality

Branch a code quality

nicolengk committed:
Add FriendlierSyntax

nicolengk committed:
Fix display issues

nicolengk committed:
Merge branch 'branch-C-FriendlierSyntax'

nicolengk committed:
Fix text wrapping issue

nicolengk committed:
Fix text alignment issue

nicolengk committed:
Add representative screenshot

nicolengk committed:
Update user guide

nicolengk committed:
Fix delete command and other messages

nicolengk committed:
Merge branch 'branch-A-UserGuide'

nicolengk committed:
Update user guide

nicolengk committed:
Fix find command message

nicolengk committed:
Fix user guide typo
